### PR TITLE
add logging for error handling edge cases

### PR DESCRIPTION
**Pull request by missinglink**
add logging for error handling edge cases

### CODE REVIEW

File: middleware/sendJSON.js [hunk 1]
```diff
@@ -42,7 +43,10 @@ function sendJSONResponse(req, res, next) {
         if( err instanceof es.errors.RequestTimeout ){ statusCode = Math.max( statusCode, 408 ); }
         else if( err instanceof es.errors.NoConnections ){ statusCode = Math.max( statusCode, 502 ); }
         else if( err instanceof es.errors.ConnectionFault ){ statusCode = Math.max( statusCode, 502 ); }
-        else { statusCode = Math.max( statusCode, 500 ); }
+        else {
+          logger.warn( 'unknown geocoding error object:', err.constructor.name, err );
```
Comment (dianashk) [inline middleware/sendJSON.js]: i thought we were going to check for existance of err.contructor here? seems a bit optimistic to assume it's there

Comment (missinglink) [inline middleware/sendJSON.js]: I had a think about it and the `if( err instanceof Error ){` check above guarantees that `err.constructor.name` will always exist.

File: middleware/500.js [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
 function middleware(err, req, res, next) {
+
   logger.error( 'Error: `%s`. Stack trace: `%s`.', err, err.stack );
-  res.header('Cache-Control','public');
-  var error = (err && err.message) ? err.message : err;
 
-  if( res.statusCode < 400 ){ res.status(500); }
-  res.json({ error: typeof error === 'string' ? error : 'internal server error' });
+  if( res.statusCode < 400 ){
+    logger.info( 'status code changed from', res.statusCode, 'to 500' );
```
Comment (missinglink) [inline middleware/500.js]:
```
console.log( "doesn't", "it", "add", "them", "for", "me?" );
doesn't it add them for me?
```

File: middleware/sendJSON.js (hunk 1 above)
Comment (missinglink) [inline middleware/sendJSON.js]:
```
var err = new Error()

err.constructor.name;
"Error"

delete err.constructor
true

err.constructor.name;
"Error"
```

File: middleware/500.js (hunk 2 above)
Comment (trescube) [inline middleware/500.js]: Ah, I learn something new about javascript every day!

File: middleware/sendJSON.js (hunk 1 above)
Comment (missinglink) [inline middleware/sendJSON.js]:
```
function MyError(){}
MyError.constructor = Error.Constructor;

var err = new MyError()

err.constructor.name;
"MyError"
```

Comment (missinglink) [inline middleware/sendJSON.js]: for some reason.. even with a typo the above works :P

Comment (missinglink) [inline middleware/sendJSON.js]: oh right, it works for any function:

```
function test(){}

test.constructor.name;
"Function"
```

Comment (missinglink) [inline middleware/sendJSON.js]: and because javascript; everything apparently

```
(99).constructor.name
"Number"
```

Comment (missinglink) [inline middleware/sendJSON.js]: oh javascript...

```
(NaN).constructor.name
"Number"
```

**Comment by orangejulius** [comment]
Superseded by https://github.com/pelias/api/pull/751

**Pull request closed without merging**